Automattic/pocket-casts-android
# Task/engage sign in card

## Description

This adds user management cluster to the Engage SDK integration.

## Testing Instructions

> [!note]
> Use the [Engage SDK verification app](https://developer.android.com/guide/playcore/engage/workflow) to test some parts of this.

1. Test this with the `debugProd` variant.
2. Open the app from clean installation.
3. Minimize the app.
4. Open the verification app and find the `ENGAGEMENT_CLUSTER`.
5. Tap on the cluster.
6. It should open the sign in screen.
7. Verify that different analytic events for signing in use `"flow": "engage_sdk"` property.
8. Sign in.
9. Go back to the verification app.
10. `ENGAGEMENT_CLUSTER` should no longer be present.
11. Go back to the Pocket Casts app.
12. Sign out.
13. Go back to the verification app.
14. `ENGAGEMENT_CLUSTER` should be present.

## Checklist
 
#### I have tested any UI changes...
